presto-native-execution/presto_cpp/main/tests/SessionPropertiesTest.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <gtest/gtest.h>
+
+#include "presto_cpp/main/SessionProperties.h"
+
+using namespace facebook::presto;
+
+class SessionPropertiesTest : public testing::Test {};
+
+TEST_F(SessionPropertiesTest, validateProperty) {
+  const std::unordered_map<std::string, std::shared_ptr<SessionProperty>> map = {
+      {"legacy_timestamp",
+       std::make_shared<SessionProperty>(
+           "legacy_timestamp",
+           "Native Execution only. Use legacy TIME & TIMESTAMP semantics. Warning: "
+           "this will be removed",
+           "BOOLEAN",
+           false,
+           "adjust_timestamp_to_session_timezone",
+           "true")},
+      {"driver_cpu_time_slice_limit_ms",
+       std::make_shared<SessionProperty>(
+           "driver_cpu_time_slice_limit_ms",
+           "Native Execution only. The cpu time slice limit in ms that a driver thread. "
+           "If not zero, can continuously run without yielding. If it is zero, then "
+           "there is no limit.",
+           "INTEGER",
+           false,
```

Comment:
And types with default value.

Reply:
Removed validation of types, since there is no helper function in `SessionProperty` to retrieve property type.